src/test/java/core/basesyntax/report/ReportFormatterImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.report;
+
+import core.basesyntax.fruitoperation.Operation;
+import core.basesyntax.fruitoperation.OperationBalance;
+import core.basesyntax.fruitoperation.OperationPurchase;
+import core.basesyntax.fruitoperation.OperationReturn;
+import core.basesyntax.fruitoperation.OperationSupply;
+import core.basesyntax.fruitoperation.Operations;
+import core.basesyntax.fruitoperation.strategy.OperationStrategy;
+import core.basesyntax.fruitoperation.strategy.OperationStrategyImpl;
+import core.basesyntax.workwithfiles.DataWriter;
+import core.basesyntax.workwithfiles.impl.FileReader;
+import core.basesyntax.workwithfiles.impl.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import junit.framework.TestCase;
+
+public class ReportFormatterImplTest extends TestCase {
+    private static ReportFormatter reportFormatter;
+
+    @Override
+    public void setUp() throws Exception {
+        Map<Operations, Operation> operationMap = new HashMap<>();
+        operationMap.put(Operations.B, new OperationBalance());
+        operationMap.put(Operations.P, new OperationPurchase());
+        operationMap.put(Operations.R, new OperationReturn());
+        operationMap.put(Operations.S, new OperationSupply());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationMap);
+        reportFormatter = new ReportFormatterImpl(operationStrategy);
+        super.setUp();
+    }
+
+    public void test1ReportFormatting_Ok() {
+        String actual = reportFormatter.createReport(
+                new FileReader("src/main/resources/test1_correct.csv"));
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator() + "banana,152";
+        assertEquals(expected, actual);
+    }
+
+    public void test2ReportFormatting_Ok() {
+        String actual = reportFormatter.createReport(
+                new FileReader("src/main/resources/test2_correct.csv"));
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator() + "banana,152";
+        assertEquals(expected, actual);
+    }
+
+    public void testReportFormattingInFile_Ok() {
+        DataWriter dataWriter = new FileWriter();
+        dataWriter.writeToFile(reportFormatter.createReport(
+                new FileReader("src/main/resources/test2_correct.csv")),
+                "src/main/resources/ApplesAndBananas");
```

Comment:
let's test this behaviour in a separate class and write some hardcoded values to the file
we should test different services independently

Reply:
done